File: loglib/src/main/java/com/robog/loglib/Schedulers.java
package com.robog.loglib;

import java.util.concurrent.Executor;
import java.util.concurrent.Executors;

/**
 * Created by yuxingdong on 2018/6/23.
 */
final class Schedulers {

    private static final Executor IO_EXECUTOR = Executors.newSingleThreadExecutor();

    static void io(Runnable runnable) {
        IO_EXECUTOR.execute(runnable);
    }
}

File: loglib/src/main/java/com/robog/loglib/Schedulers.kt
package com.robog.loglib

import java.util.concurrent.Executors

/**
 * Created by yuxingdong on 2018/6/29.
 */
private val IO_EXECUTOR = Executors.newSingleThreadExecutor()
private val WORK_EXECUTOR = Executors.newCachedThreadPool()

fun io(f: () -> Unit) {
    IO_EXECUTOR.execute(f)
}

fun work(f: () -> Unit) {
    WORK_EXECUTOR.execute(f)
}